### DISCUSSION TITLE
Maui Preprocessor in Blazor Hybrid

### DISCUSSION THREAD

**Discussion by AathifMahir** [dotnet/maui#19633] [Q&A]
I have a Blazor Hybrid App, and all the Presentation/Razor Components reside in a Class Library that can be shared between the Blazor Web Assembly app and Blazor Hybrid App. The current issue is that I need to implement specific logic that runs exclusively on the Blazor Hybrid App within that class library. How can I accomplish this? Is there a preprocessor for Maui or any other solution?

For example:

#if Maui

**Comment by foximoxi** [comment]
You can achieve that using Dependency Injection.
1. Define service interface in your shared lib and desired implementation in target project (MAUIHybrid/Web or both).
You can still use common classes shared between projects.
2. Register e.g. builder.Services.AddTransient<IMyService,MAUIMyService>();

**Comment by AathifMahir** [comment]
> You can achieve that using Dependency Injection.
> 1. Define service interface in your shared lib and desired implementation in target project (MAUIHybrid/Web or both).
> You can still use common classes shared between projects.
> 2. Register e.g. builder.Services.AddTransient<IMyService,MAUIMyService>();
> 
> 
> 

In my case, I don't have different implementation for web and hybrid project, most of the implementation were same except I want add some additionally code to razor components when running on hybrid, that's all

**Comment by foximoxi** [comment]
If you need just a flag which tells you that your code is executed in Hybrid you can create static variable in common lib
`public static class Config
{
 public static bool IsMaui {get;set;}
}`

Than you can set Config.IsMaui=true on your MAUI project during initialization
Your components can use this flag for specific logic/component rendering

**Reply by AathifMahir** [reply]
> If you need just a flag which tells you that your code is executed in Hybrid you can create static variable in common lib
> `public static class Config
> {
>  public static bool IsMaui {get;set;}
> }`
> 
> Than you can set Config.IsMaui=true on your MAUI project during initialization
> Your components can use this flag for specific logic/component rendering

That's a great idea, in that case I can extend the functionality with which platform that app is running on as well. Thanks
